File: app/src/main/java/org/test/kotlin/demo/repos/search/SearchReposFragment.java
package org.test.kotlin.demo.repos.search;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.textfield.TextInputLayout;

import org.test.kotlin.demo.R;
import org.test.kotlin.demo.api.dto.RepositoryDTO;
import org.test.kotlin.demo.repos.ReposAdapter;
import org.test.utils.LoadingAdapter;

import java.util.List;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;
import dagger.android.support.DaggerFragment;

import static org.test.utils.ViewUtils.hideKeyboard;

public class SearchReposFragment extends DaggerFragment implements SearchReposContract.View {

    @Inject
    SearchReposContract.Presenter presenter;

    public static SearchReposFragment create() {
        return new SearchReposFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_search, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        TextInputLayout query = view.findViewById(R.id.query);
        EditText editText = query.getEditText();
        editText.setOnEditorActionListener((v, a, e) -> {
            performSearch();
            return true;
        });

        FloatingActionButton search = view.findViewById(R.id.search);
        search.setOnClickListener($ -> performSearch());
    }

    private void performSearch() {
        View view = getView();
        hideKeyboard(view);

        TextInputLayout query = view.findViewById(R.id.query);
        query.setErrorEnabled(false);
        String queryValue = query.getEditText().getText().toString();
        presenter.onQueryChange(queryValue);

        RecyclerView recyclerView = view.findViewById(R.id.recycler);
        recyclerView.setAdapter(new LoadingAdapter());
    }

    @Override
    public void showRepositories(@NonNull List<RepositoryDTO> repositories) {
        View view = getView();

        RecyclerView recyclerView = view.findViewById(R.id.recycler);
        recyclerView.setAdapter(new ReposAdapter(repositories));
    }

    @Override
    public void showError(Throwable throwable) {
        View view = getView();

        TextInputLayout query = view.findViewById(R.id.query);
        query.setError(throwable.getLocalizedMessage());

        RecyclerView recyclerView = view.findViewById(R.id.recycler);
        recyclerView.setAdapter(null);
    }

}

File: app/src/main/java/org/test/kotlin/demo/repos/search/SearchReposFragment.kt
package org.test.kotlin.demo.repos.search

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import dagger.android.support.DaggerFragment
import kotlinx.android.synthetic.main.fragment_recycler.recycler
import kotlinx.android.synthetic.main.fragment_search.*
import org.test.kotlin.demo.R
import org.test.kotlin.demo.api.dto.RepositoryDTO
import org.test.kotlin.demo.repos.ReposAdapter
import org.test.utils.LoadingAdapter
import org.test.utils.hideKeyboard
import org.test.utils.inflate
import org.test.utils.log
import javax.inject.Inject

class SearchReposFragment : DaggerFragment(), SearchReposContract.View {

    @Inject
    internal lateinit var presenter: SearchReposContract.Presenter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?) =
            container!!.inflate(R.layout.fragment_search)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        query.editText!!.setOnEditorActionListener { _, _, _ ->
            performSearch()
            true
        }

        search.setOnClickListener { performSearch() }
    }

    private fun performSearch() {
        view!!.hideKeyboard()

        query.isErrorEnabled = false
        val queryValue = query.editText!!.text.toString()
        presenter.onQueryChange(queryValue)

        recycler.adapter = LoadingAdapter()
    }

    override fun showRepositories(repositories: List<RepositoryDTO>) {
        recycler.adapter = ReposAdapter(repositories)
    }

    override fun showError(throwable: Throwable) {
        throwable.log()

        query.error = throwable.localizedMessage

        recycler.adapter = null
    }

    companion object {

        fun create() = SearchReposFragment()

    }

}
